tracer/src/Datadog.Trace/Configuration/ConfigurationSources/CompositeConfigurationSource.cs

```diff
@@ -41,34 +37,18 @@ public CompositeConfigurationSource(IEnumerable<IConfigurationSource> sources)
         /// <param name="source">The configuration source to add.</param>
         public void Add(IConfigurationSource source)
         {
-            if (source == null) { ThrowHelper.ThrowArgumentNullException(nameof(source)); }
+            if (source == null!) { ThrowHelper.ThrowArgumentNullException(nameof(source)); }
```

Comment:
this `null!` is funny, why is it needed here 🤔

Reply:
Yeah 😅 It's because the nullable reference types indicate that `source` _can't_ be `null`, so it's to stop a warning. Once we have null reference types everywhere, we can remove the arg check completely, but for now, it's safest to keep it! 😄